python-package/securexgboost/rpc/protos/remote.proto

```diff
@@ -129,15 +149,29 @@ message DMatrixAttrs {
     repeated string filenames = 1;
     repeated string usernames = 2;
     uint32 silent = 3;
-    string username = 4;
-    numproto.protobuf.NDArray signature = 5;
-    uint32 sig_len = 6;
+    string username = 4; // [deprecated] moved to DMatrixAttrsRequest
+}
+
+// Wrapper around DMatrixAttrs to include sequence number
+message DMatrixAttrsRequest {
+    DMatrixAttrs attrs = 1;
```

Comment:
Any reason we wrap the old RPC message into a new one (`*Request`) instead of just adding the new parameters to the old message?

Reply:
it made more sense to me personally

I don't think sequence numbers and signatures should be part of "dmatrix attributes" or "booster attributes" for example. it made more sense to make them part of the idea of an RPC request specifically which is why I made a "request" protobuf for each